testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/principaltransformers/CasePrincipalTransformerTestCase.java

```diff
@@ -0,0 +1,181 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.wildfly.test.integration.elytron.principaltransformers;
+
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static javax.servlet.http.HttpServletResponse.SC_UNAUTHORIZED;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.api.ServerSetupTask;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.test.integration.management.util.CLIWrapper;
+import org.jboss.as.test.integration.security.common.Utils;
+import org.jboss.as.test.integration.security.common.servlets.SimpleSecuredServlet;
+import org.jboss.as.test.integration.security.common.servlets.SimpleServlet;
+import org.jboss.as.test.shared.ServerReload;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.test.security.common.elytron.PropertyFileBasedDomain;
+
+
+/**
+ * Test case for 'case-principal-transformer' Elytron subsystem resource.
+ *
+ * @author Sonia Zaldana Calles <szaldana@redhat.com>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({CasePrincipalTransformerTestCase.SetupTask.class})
+public class CasePrincipalTransformerTestCase {
+
+    private static final String DEP_SECURITY_DOMAIN = "case-principal-domain";
+    private static final String USER = "USER1";
+    private static final String PASSWORD = "password1";
+    private static final String ROLE = "JBossAdmin";
+    @ArquillianResource
+    private static ManagementClient client;
+
+    @Deployment(name = DEP_SECURITY_DOMAIN)
+    public static WebArchive createDeployment() {
+        final WebArchive war = ShrinkWrap.create(WebArchive.class, DEP_SECURITY_DOMAIN + ".war");
+        war.addClasses(SimpleServlet.class);
+        war.addClasses(SimpleSecuredServlet.class);
+        war.addAsWebInfResource(ConstantPrincipalTransformerTestCase.class.getPackage(), "principal-transformer-web.xml", "web.xml");
+        war.addAsWebInfResource(Utils.getJBossWebXmlAsset(DEP_SECURITY_DOMAIN), "jboss-web.xml");
+        return war;
+    }
+
+    /**
+     * Test whether name of user passed to security domain is adjusted into upper case with
+     * the 'case-principal-transformer'. Test also checks that authentication passes for correct password.
+     */
+    @Test
+    @OperateOnDeployment(DEP_SECURITY_DOMAIN)
+    public void testPassingExistingUser(@ArquillianResource URL webAppURL) throws Exception {
```

Comment:
It would also be good to have a test that makes use of a `case-principal-transformer` that sets `upper-case` to `false`. 

Just a minor comment but it would also be good to give these tests names that indicate what the test is checking - e.g., `testCasePrincipalTransformerUpperCase`

Reply:
fixed